### ISSUE TITLE
The test system could also use some :heart:

### ISSUE THREAD

**Issue by aaronmondal** [tensorflow/text#719]
There is currently a lack of information on how to test the repository. Also it would be nice to have some way of seeing which tests are supposed to pass and which ones are currently expected to fail.

Running `oss_scripts/run_tests.sh` does not work as expected and is currently not documented. It runs tests in parallel and fails tests at random (roughly **19/36**). Note that adding `--jobs=1` reduces that number to only these failures on my machine:
```bash
//tensorflow_text:whitespace_tokenizer_test                    FAILED TO BUILD
//tensorflow_text:bert_tokenizer_test                          FAILED in 14.7s
//tensorflow_text:gather_with_default_op_test                  FAILED in 5.4s
//tensorflow_text:sentence_breaking_ops_test                   FAILED in 5.9s
```
Note that the benchmark targets `tokenizers_benchmark` and `ops_benchmark` may not be intended to be in there.

Now the problem is that it is unclear whether these tests are expected to fail or the failures come from my setup as there is no official "Hey these tests are expected to fail".
Another issue is e.g. that many more of these test fail when not running with `--jobs=1`.

The failure for `whitespace_tokenizer` seems to be expected as some recent commits seem to change that a lot, but I am not sure about the other ones and ill look into it.

**To be clear, i do not think it matters much whether these tests fail in general, but it makes it very hard to change things up when there is no information on which tests are currently expected to pass/fail.**

---

For anyone interested in reproducing the following commands should work (same as `oss_scripts/run_tests.sh`, but singlethreaded and ignoring failures):
```sh
# Generate a .bazelrc
./oss_scripts/configure.sh

# Run the tests
bazel test --test_output=errors --keep_going --jobs=1 tensorflow_text:all
```
Note that for the benchmarks to pass you need `tensorflow_datasets` in your python toolchain (e.g. via `pip install tensorflow_datasets`). Also note that running the benchmarks like that may be pointless anyways, but lets just ignore that for now :tada:

**Comment by aaronmondal** [comment]
Ok upon rebuilding and retesting I think these are the fixes required:
- [ ] `bert_tokenizer_test` see https://github.com/TensorSpeech/TensorFlowASR/issues/71#issue-760832910 . Should be straightforward to fix. Probably only occurs on setups with GPUs.
- [ ] `sentence_breaking_ops_test` same as above. Probably only occurs on setups with GPUs.
- [ ] `gather_with_default_op_test` fails the `testIndexOutOfRange`. Not sure abour this one.

Around 15 other tests fail when running in parallel. This is because running parallel tests on a single GPU is probably a bad idea :dancers:. Probably fluctuates by GPU model and GPU memory. Easily fixed by setting `--jobs=1`.

Note that some other tests will fail if depedencies like `tensorflow_datasets` and `tensorflow_estimator` are missing. This is setup specific and a non-issue because lets be realistic here anyone who builds TFT will have every other TF package anyways :shrug:. There may be a way to prevent those failures though by adding the respective packages to the test target depencencies.

**Comment by broken** [comment]
I agree the tests system can definitely use some work. Let me start by saying that all of our tests are currently passing. These are run on all platforms (linux, mac, win) a few times a day, and are also ran for each PR before it is merged.

This is the Kokoro check you see in the PR. Unfortunately, only whether it passed or failed is public and the detailed results are not - not because we don't want them to be, but because it's an internal system and we haven't set anything up to mirror the results. (nobody has asked about it, and so I think we just forgot this needed to happen). 

We continually monitor that the tests work in this automated environment, but nobody on our team really runs them outside of it (you need to build TF to run them - see #160), so I'm not that surprised you ran across issues.

If you could create a PR adding the necessary flags to `run_tests.sh`, that would be great. Our test framework doesn't actually call that file, so any changes to it won't affect them.
